src/components/Icon/Icon.tsx

```diff
@@ -4,16 +4,44 @@ import { customPropTypes, UIComponent, createShorthandFactory } from '../../lib'
 import { IconBehavior } from '../../lib/accessibility/'
 
 import svgIcons from './svgIcons'
+import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+import { Extendable } from '../../../types/utils'
 
 export type IconXSpacing = 'none' | 'before' | 'after' | 'both'
+export type IconSize =
+  | 'micro'
+  | 'mini'
+  | 'tiny'
+  | 'small'
+  | 'normal'
+  | 'large'
+  | 'big'
+  | 'huge'
+  | 'massive'
+
+export interface IIconProps {
+  as?: any
+  bordered?: boolean
+  circular?: boolean
+  className?: string
+  disabled?: boolean
+  font?: boolean | string
+  name?: string
+  size?: IconSize
+  svg?: boolean
+  xSpacing?: IconXSpacing
+  accessibility?: object | Function
```

Comment:
just as a side note - we should introduce it as a type consumed from `accessibility` module

Reply:
Sure, I was defining now the accessibility exactly as it is describe in the PropTypes, once we add interface for it, we can easily update this prop across all components.